cli/configssh.go

```diff
@@ -507,7 +538,19 @@ func (r *RootCmd) configSSH() *clibase.Cmd {
 			Flag:        "ssh-host-prefix",
 			Env:         "",
 			Description: "Override the default host prefix.",
-			Value:       clibase.StringOf(&userHostPrefix),
+			Value:       clibase.StringOf(&sshConfigOpts.userHostPrefix),
+		},
+		{
+			Flag:        "wait",
+			Env:         "CODER_CONFIGSSH_WAIT", // Not to be mixed with CODER_SSH_WAIT.
+			Description: "Set the option to wait for the the startup script to finish executing. This is the default if the template has configured the agent startup script behavior as blocking. Can not be used together with --no-wait.",
+			Value:       clibase.BoolOf(&sshConfigOpts.wait),
+		},
+		{
+			Flag:        "no-wait",
+			Env:         "CODER_CONFIGSSH_NO_WAIT", // Not to be mixed with CODER_SSH_NO_WAIT.
+			Description: "Set the option to enter workspace immediately after the agent has connected. This is the default if the template has configured the agent startup script behavior as non-blocking. Can not be used together with --wait.",
+			Value:       clibase.BoolOf(&sshConfigOpts.noWait),
```

Comment:
Should we deprecate one of these options?

Reply:
It's a valid thing to point out, however, both are needed because depending on the agent configuration, you may want to use the opposite of what is configured.

If we only used `--wait` and `--wait=false`, for instance, we would still need to detect the presence of this flag. I.e. we can't simply check the boolean, we have to check if the flag/env was provided so that we know whether or not to override the template. We also can't say on the CLI what the default value for this flag is because it's workspace dependent.

I'm not sure clibase easily provides this info and I also think it's bad UX to require `=false` (or true). 🤔

For most clarity, this should perhaps be `--startup-script-behavior=auto|blocking|non-blocking`, but again, not very UX friendly.

Reply:
It's a great suggestion. What do you think @bpmct?

Reply:
Tried implementing it here: [`5e8c487` (#7894)](https://github.com/coder/coder/pull/7894/commits/5e8c487c482d1f5d3b3db3d62fb1a97959950494), I like it.

Reply:
I like this too.